lms/djangoapps/instructor/tests/views/test_instructor_dashboard.py

```diff
@@ -135,6 +140,90 @@ def has_instructor_tab(user, course):
+    )
+    @ddt.unpack
+    def test_discussion_tab_staff(self, access_role, is_pages_and_resources_enabled,
+                                  is_legacy_discussion_setting_enabled, is_discussion_tab_available):
+        """
+        Verify that the Discussion is visible/hidden with flags enabled/disabled for course for staff role.
+        """
+        discussion_section = '<li class="nav-item"><button type="button" class="btn-link discussions_management" ' \
+                             'data-section="discussions_management">Discussions</button></li>'
+
+        with override_waffle_flag(ENABLE_PAGES_AND_RESOURCES_MICROFRONTEND, is_pages_and_resources_enabled):
+            with override_waffle_flag(OVERRIDE_DISCUSSION_LEGACY_SETTINGS_FLAG, is_legacy_discussion_setting_enabled):
+                user = UserFactory.create()
+                CourseAccessRoleFactory(
+                    course_id=self.course.id,
+                    user=user,
+                    role=access_role,
+                    org=self.course.id.org
+                )
+                set_course_cohorted(self.course.id, True)
+                self.client.login(username=self.user.username, password='test')
+                response = self.client.get(self.url).content.decode('utf-8')
+                self.assertEqual(discussion_section in response, is_discussion_tab_available)
+
+    @ddt.data(
+        ('instructor', False, False, True),
+        ('instructor', True, False, False),
+        ('instructor', True, True, True),
+    )
+    @ddt.unpack
+    def test_discussion_tab_instructor(self, access_role, is_pages_and_resources_enabled,
+                                       is_legacy_discussion_setting_enabled, is_discussion_tab_available):
+        """
+        Verify that the Discussion is visible/hidden with flags enabled/disabled for course for instructor role.
+        """
+        discussion_section = '<li class="nav-item"><button type="button" class="btn-link discussions_management" ' \
+                             'data-section="discussions_management">Discussions</button></li>'
+
+        with override_waffle_flag(ENABLE_PAGES_AND_RESOURCES_MICROFRONTEND, is_pages_and_resources_enabled):
+            with override_waffle_flag(OVERRIDE_DISCUSSION_LEGACY_SETTINGS_FLAG, is_legacy_discussion_setting_enabled):
+                user = UserFactory.create()
+                CourseAccessRoleFactory(
+                    course_id=self.course.id,
+                    user=user,
+                    role=access_role,
+                    org=self.course.id.org
+                )
+                set_course_cohorted(self.course.id, True)
+                self.client.login(username=self.user.username, password='test')
+                response = self.client.get(self.url).content.decode('utf-8')
+                self.assertEqual(discussion_section in response, is_discussion_tab_available)
+
+    @ddt.data(
+        ('global_staff', False, False, True),
```

Comment:
This `global_staff` is not changing with data, so this is not needed as a part of `ddt.data`.